torch_xla/csrc/tensor.cpp

```diff
@@ -840,7 +847,21 @@ bool XLASymNodeImpl::bool_() {
   return dn->getDynamicValue() != 0;
 }
 
-bool XLASymNodeImpl::has_hint() { return true; }
+// "a SymInt has_hint" is equivalent to "a SymInt is backed". Unbacked SymInt is
+// the result of a data dependent output like nonzero; we don't know what the
+// value is because it's data dependent.
+// xw32: what's the justification that this function can always return false?
+// What if we have
+/*
+t0=torch.zeros(3)
+t2=torch.nonzero(t1)
+sz=t2.shape[0]+t0.shape[0]
+*/
+// then sz is always an unbacked SymInt.
+bool XLASymNodeImpl::has_hint() {
+  std::cout << "xw32, file=" << __FILE__ << ", line=" << __LINE__ << "function=" << __FUNCTION__ << ": " << std::endl;
+  return false;
```

Comment:
hi @ezyang , just want to confirm if my understanding is correct. The reason why we can always return false for `XLASymNodeImpl::has_hint` is that torch_xla only create a `SymNodeImpl` for `nonzero` output, such as [here](https://github.com/pytorch/xla/blob/8e1d3ae98f644f5c05645342d8408d0553fa62ba/torch_xla/csrc/tensor_impl.cpp#L191-L195). Moreover, during propagation such as in 
```
t0=torch.zeros(3)
t2=torch.nonzero(t1)
sz=t2.shape[0]+t0.shape[0]
```
`sz` will remain representing a unbacked SymInt. So it's ok to always return false. Does it sound good to you?

Reply:
yes